### DISCUSSION TITLE
24 September Dev Session

### DISCUSSION THREAD

**Discussion by tpike3** [mesa/mesa#1411] [General]
When: 24 Sept---- 2000 UTC+8/1300 CET/1200 GMT/0800 ET

Where: [via Matrix](https://matrix.to/#/#mesa-dev:matrix.org)

Proposed Agenda:

- Get consensus on #1406 
- Mesa - Geo
- Discuss Sugarscape Tutorial 

@jackiekazil @rht @Corvince @wang-boyu @aiwhoo @nunezco2 Please feel free to add or takeaway from this agenda

**Comment by rht** [comment]
I want to hear the story behind the uncovering the mystery of the trade volume in Sugarscape with trading. And the progress with the remaining problem: carrying capacity discrepancy.

**Comment by rht** [comment]
I wonder if we can get a consensus on https://github.com/projectmesa/mesa/discussions/1406 and move on this month.

**Comment by wang-boyu** [comment]
Adding https://github.com/projectmesa/mesa-geo/pull/101 for further discussions.

**Comment by jackiekazil** [comment]
Are there any notes from this session?

**Reply by tpike3** [reply]
Sorry about that I will get them posted this weekend,,, the tutorial is giving me some unexpected challenges.

**Comment by jackiekazil** [comment]
No worries.

**Comment by tpike3** [comment]
## 24 Sep Dev Session Notes
---------------------------------------

Apologies for the delay. Due to delay and a faulty memory, if I mispresented anything please let me know in a reply and I will be happy to correct. 

### Item 1: 

#1406 Based on discussions the goal is to provide user options of examples based on their use/need. The selected way forward is a tiered system ---

1. Include seminal models that demonstrate a key functionality in examples folder. 
2. Other validated models that demonstrate key functionalities of mesa in a separate repository. One question remained from the discussion so please vote in reply below or by messaging me in Matrix (apologies couldn't figure out how to do this anonymized-- I am open to suggestions).
3. Have a linked README in separate examples repository to larger models that use Mesa.  

### Vote:  Where should we store videos of the models? 
- Option 1: Store in youtube channel potentially in a series of playlist. See [Complexity Explorer Example](https://www.youtube.com/user/ComplexityExplorer/playlists) 
- Option 2: Videos stored in github repo of videos

### Item 2:

Mesa - Geo: Challenge of PyPI test access. Debated whether or not just to set up a different test PyPI access, while we work to regain access to test Mesa Geo. However, Mesa-Geo 0.3.0 was released this week! Thanks @wang-boyu 


### Item 3: 

Tutorial: Proceeding slowly but surely. One item not brought up is prior to filiming the later part of the tutorial (e.g. session 15), we need to do a release. Currently the tutorial requires `self.running = True` which is no longer the case

**Reply by jackiekazil** [reply]
Update on PyPI Test access -- I have been chatting w/ @Corvince. He is working on making it happen, but there are some delays. Hopefully resolved soon.

**Reply by wang-boyu** [reply]
Special thanks to @jackiekazil on guiding me through the whole process to upload the new version of Mesa-Geo to PyPI : )

**Comment by rht** [comment]
Option 1: Store in youtube channel potentially in a series of playlist. Cast your vote by upvoting this message.

**Reply by aiwhoo** [reply]
It may be good to just embed the videos (YouTube link) onto a (the GitHub) site.

**Reply by rht** [reply]
Note that an Agents.jl maintainer said creating and uploading those YouTube videos whenever the tutorials are updated, is cumbersome.

**Reply by wang-boyu** [reply]
Another issue is that GitHub doesn't support embedded YouTube videos in README files. We could only link to them (for example using a picture).

It is possible to embed YouTube videos using Read the Docs. So we may need to maintain two different ways.

**Comment by rht** [comment]
Option 2: Videos stored in github repo of videos. Cast your vote by upvoting this message.

**Reply by jackiekazil** [reply]
with this option I have a concern about the size of the video in a repo and how that might bloat a repo.

**Reply by rht** [reply]
If repo size is a concern, we can always truncate the number of commits in the repo to be at most 20 or something. Old commits can be discarded.

**Comment by wang-boyu** [comment]
Not sure if it's possible, but how about a 3rd option where we implement https://github.com/projectmesa/mesa/issues/885, and add a video generation step into the doc building procedure? This probably works only for Read the Docs though.

**Reply by rht** [reply]
The choice of the way the videos are generated is orthogonal to where they should be stored. And yes, ideally if the video generation can be automated like Agents.jl devs do it.

**Reply by wang-boyu** [reply]
Oh sorry what I meant is (if it's possible) to generate videos on-the-fly while generating documents, so we don't need to store them at YouTube or GitHub repo.
